apps/frontend/src/pages/EditProductPage.jsx

```diff
@@ -0,0 +1,309 @@
+import { useState, useEffect } from "react";
+import { Formik, Form } from "formik";
+import * as Yup from "yup";
+import InputField from "../components/InputField";
+import Button from "../components/Button";
+import GenericList from "../components/GenericList";
+import { MdAdd, MdOutlineCancel, MdSave } from "react-icons/md";
+import { FaMinus } from "react-icons/fa";
+import { getProductById, updateProduct, updateProductCategories } from "../services/ProductService";
+import { createCategory, getAllCategories } from "../services/CategoryService";
+import { useNavigate, useParams } from "react-router-dom";
+import { toast } from "sonner";
+import { useUser } from "../hooks/UserUser";
+
+const EditProductSchema = Yup.object().shape({
+  name: Yup.string().required("El nombre del producto es obligatorio"),
+  buyPrice: Yup.number()
+    .required("El precio de compra es obligatorio")
+    .min(1, "Debe ser mayor o igual a 1"),
+  sellPrice: Yup.number()
+    .required("El precio de venta es obligatorio")
+    .min(1, "Debe ser mayor o igual a 1"),
+  lowStock: Yup.number()
+    .min(0, "Debe ser mayor o igual a 0")
+    .notRequired(),
+  lowStockEnabled: Yup.boolean(),
+});
+
+function EditProductPage() {
+  const { id } = useParams();
+  const user = useUser();
```

Comment:
It should be {user} so that you are using only the user from the context and not the user and the editUser